pkg/kubelet/kubelet.go

```diff
@@ -2737,6 +2747,145 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}, 100*time.Millisecond, stopCh)
 }
 
+// handleMaxContainerCheckpoints goes through the list of checkpoints of a
+// container in a pod and deletes old checkpoints if there are more than
+// 'kl.maxContainerCheckpoints'. To track the number of checkpoints a JSON file
+// is created and updated for each new checkpoint. If a new checkpoint would
+// result in more than 'kl.maxContainerCheckpoints' the oldest checkpoint will
+// be removed.
+func (kl *Kubelet) handleMaxContainerCheckpoints(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+	checkpointTime int64,
+) error {
+
+	if kl.maxContainerCheckpoints < 0 {
+		// This cannot really happen as the kubelet config parser will
+		// set this to 10 as a default value if 0 or not set. The error
+		// handling is mainly here to make sure tests are working.
+		return errors.New("maxContainerCheckpoints needs to be >= 0")
+	}
+
+	checkpointJsonName := fmt.Sprintf(
```

Comment:
This is taking a hard dependency on an external library and config file format we don't control ... it's not clear that we should take that dependency, or what the compatibility guarantees of that file are across kubelet versions.

Reply:
@liggitt Thank you for pointing this out! Adrian is a maintainer of the [`checkpointctl`](https://github.com/checkpoint-restore/checkpointctl) library. This library enables using the same checkpoint config format across repositories such as [CRI-O](https://github.com/cri-o/cri-o/blob/50e274e807d8d2439d1f90af688614e666cf41d6/server/container_checkpoint.go#L6) and [Podman](https://github.com/containers/podman/blob/8457bb55428cad65a60a2c3610caa12dba5b2049/pkg/checkpoint/checkpoint_restore.go#L9).

What do we need to do to ensure compatibility guarantees across kubelet versions?